Upstream: https://github.com/mesonbuild/meson/pull/9211

Fix the following error when configuring pango (or any project with
appleframeworks listed as a dependency) using GCC:

ERROR: Dependency "appleframeworks" not found, tried framework
--- mesonbuild/compilers/mixins/clike.py.orig	2021-09-02 16:07:01.000000000 -0400
+++ mesonbuild/compilers/mixins/clike.py	2021-09-02 16:07:36.000000000 -0400
@@ -1199,9 +1199,6 @@
         Finds the framework with the specified name, and returns link args for
         the same or returns None when the framework is not found.
         '''
-        # TODO: maybe this belongs in clang? also, should probably check for macOS?
-        if self.id != 'clang':
-            raise mesonlib.MesonException('Cannot find frameworks with non-clang compiler')
         return self._find_framework_impl(name, env, extra_dirs, allow_system)
 
     def get_crt_compile_args(self, crt_val: str, buildtype: str) -> T.List[str]:
